### ISSUE TITLE
HTTP fetch re-ordering requested parameters

### ISSUE THREAD

**Issue by chipnetics** [vlang/v#13397]
**V version:** 
V full version: V 0.2.4 c1711b8
**OS:** 
OS: linux, Debian GNU/Linux 11 (bullseye)

**What did you do?**

Making an http.fetch request to an API endpoint will dissasemble the passed url parameters, and re-assemble using a sorted map.  The problem is, I have come across an API that expects (in the below example) the signature parameter to be **passed last**.  This is a financial trading API, so it's likely that to reduce server-side overhead parsing parameters, they expect the signature last at all times.

result := http.fetch(
  	  	url: "https://someapi.com/endpoint?gamma=1&tau=2&alpha=3&signature=123",
	   	method: .post,
  	  	header: http.new_custom_header_from_map({"X-MBX-APIKEY": api_key}) or { panic(err) }
	 ) or { return Leverage{}, true }

**What did you expect to see?**

 HTTP Post with the ordering intact: https://someapi.com/endpoint?gamma=1&tau=2&alpha=3&signature=123
 
**What did you see instead?**

compiling with v test.v -d trace_http_request shows the POST is being re-ordered using a map, as:

https://someapi.com/endpoint?alpha=3&gamma=1&signature=123&tau=2

...which results in the API endpoint returning an error, as the signature is not where it is expected to be (is 2nd from last now).

_I have confirmed that the location is indeed critical by manually testing with Curl and placing the signature elsewhere, and am getting the same results as V._
 
It may be worth noting that the signature paramter in the above is an HMAC SHA256 encryption of everything prior to the parameter; which is another reason it probably has to be passed last.  It may be the case that any API requiring the POST/GET/DELETE to be encrypted may require such paramters last (and not just financial trading based APIs).

**Comment by vincenzopalazzo** [comment]
This is right, as humans we can not follow some standards. By RFC there is no requirements on the order of parameters, but some server required this order.

This is a good discussion where https://stackoverflow.com/questions/28692230/is-the-order-of-named-parameters-important-in-a-url, and I will try to update the request logic.

Thanks to reporting this
